src/ZkFold/Symbolic/Cardano/Types/Value.hs

```diff
@@ -36,17 +35,3 @@ deriving instance
 
 instance Symbolic context => Scale Natural (Value n context) where
     n `scale` Value v = Value $ fmap (\((pid, aname), q) -> ((pid, aname), n `scale` q)) v
-
-instance (Haskell.Ord (PolicyId context), Haskell.Ord (AssetName context), Symbolic context) => Semigroup (Value n context) where
```

Comment:
Why are these instances removed?

Reply:
I can put them back but to be honest they didn't make much sense to me, so I ended up removing them when making modifications. The `mempty`/`zero` value has 0 length even though `Value` is a newtype over `Vector n`. That can't be right, right?

Reply:
Yeah, now that you mention it, looks strange to me.
Pinging @vlasin to take a look at it too. Maybe Symbolic `List` was expected to be used here instead of `Vector`?

Reply:
Yes, exactly. `Value` is a "basket of assets". We should be able to merge two baskets into one. Using `Vector n` for this is less than ideal, but we don't have the Symbolic List just yet.

Reply:
But we could use functions with more sensible signatures. Shouldn’t combining add the lengths. And shouldn’t empty value be length zero?

Reply:
The most sensible signatures would be when we use Symbolic lists here instead of Vectors, for now it is better to keep instances in place for future reference when we do the migration

Reply:
(Also combining does not necessarily add the lengths, the resulting length might be smaller because of equal keys)

Reply:
Yes, for the time being, we could think of `n` as the maximal possible size of the basket. So, an "overflow" will result in a system with an empty solution set.

Reply:
Added instances back.